clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/JChannelFactoryBuilder.java

```diff
@@ -86,7 +82,11 @@ public JChannelFactoryBuilder(PathAddress address) {
         ServiceBuilder<ChannelFactory> builder = target.addService(this.getServiceName(), new ValueService<>(value))
                 .addDependency(ServerEnvironmentService.SERVICE_NAME, ServerEnvironment.class, this.environment)
                 .setInitialMode(ServiceController.Mode.ON_DEMAND);
-        Stream.concat(Stream.of(this.transport, this.relay).filter(Objects::nonNull), this.protocols.stream()).forEach(dependency -> dependency.register(builder));
+        for (ValueDependency<ProtocolConfiguration> protocol : protocols) {
+            if (protocol != null) protocol.register(builder);
+        }
+        if (transport != null) transport.register(builder);
```

Comment:
Logic in configure(...) ensures transport is never null.  Only relay is optional.